### ISSUE TITLE
Could not reproduce paper result

### ISSUE THREAD

**Issue by zhujl3** [swj0419/in-context-pretraining#5]
We used llama7b, from the deduplicated encommoncrawl dataset, and constructed a 200B training set using the concatenation method mentioned in the paper. We trained the baseline using this 200B data with random concatenation. We encountered three inconsistencies with the original paper.

1.When the training reaches 200B token, we observed that the paper's train loss, baseline is 2.0, ICLM is 1.5, which is lower by 25%; However, in our training, when the training volume reaches 200B, the baseline loss is 2.1, and ICLM loss is also 2.0, which only decreased by 5%.
![loss_train](https://github.com/swj0419/in-context-pretraining/assets/12773902/8dad04d9-0bf5-409c-b939-a79251d5c527)
![loss_paper](https://github.com/swj0419/in-context-pretraining/assets/12773902/b093d2e3-9f29-4567-a9a6-16d45828e09d)
2. When testing race-high/race-middle dataset with 2-shot，when training 200B token, the acc in both model seems to lower than 0.25, which is significantly worse than the results reported in the original paper.Also we could not find a obvious difference in other dataset used in paper such as qasper
3. we also try a continue training, but also we could not get a obvious difference in any dataset except SST2

Could you give some advice. Or tell us how you test the dataset with 2-shot?

**Comment by swj0419** [comment]
Thanks for your interest in our work. 

(1) Did you train the model from scratch or is this continue pretraining?
(2) what are the race-high/middle scores you got?

**Comment by swj0419** [comment]
We haven't explored the continue pretraining setup yet. However, the results from pretraining from scratch that you've shared appear quite unusual. Based on your result, it seems that for the four-choice classification in Race, the performance is nearly at random levels. In contrast, our evaluation of BoolQ on 7B models achieves approximately 70% accuracy. Regarding Race-high, we directly compare the probabilities of each choice, rather than generating options A, B, C, D.

**Comment by zhujl3** [comment]
Thank you for your reply !
After discussion, we speculate that the most likely differing factor comes from the variance in training data.
Your paper mentions the use of the EN-COMMONCRAWL dataset, but also give a reference to **CCNet** which indicate a data filtering pipeline.

My question is:
(1) If you filter EnCommonCrawl dataset with the method using in CCNet
(2) Could you give us the snapshot of commoncraw dataset?  Maybe the snapshot difference also has a influence : )
